### PR TITLE
[Sui native bridge] - Add update_node_url function to bridge

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

as titled

## Test plan 

How did you test the new or updated feature?

Move unit tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-framework/packages/bridge/sources/committee.move [hunk 1]
```diff
@@ -268,6 +267,19 @@ module bridge::committee {
...
+        let mut idx = 0;
+        while (idx < self.members.size()) {
+            let (_, member) = self.members.get_entry_by_idx_mut(idx);
+            if (member.sui_address == ctx.sender()) {
+                member.http_rest_url = new_url;
+                return
+            };
+            idx = idx + 1;
+        };
+        abort ESenderNotActiveValidator
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/committee.move:280]: how aboutt `ESenderIsNotInBridgeCommittee`

**Review by longbowlu — COMMENTED** [review]
the change itself looks good! 
1. @dariorussi  do you mind taking another look for higher confidence (and see if your move test lib can be leveraged here)
2. the auto-gened graphql changes scare me. I asked how to avoid it

File: crates/sui-framework/packages/bridge/tests/commitee_test.move [hunk 2]
```diff
@@ -151,6 +151,92 @@ module bridge::committee_test {
...
+
+        // Update URL should fail for validator @0xB
+        committee.update_node_url(
+            b"test url",
+            &tx(@0xB, 0),
+        );
+
+        test_utils::destroy(committee);
+        test_scenario::return_shared(system_state);
+        test_scenario::end(scenario);
```
Comment (dariorussi) [inline crates/sui-framework/packages/bridge/tests/commitee_test.move:237]: just fyi you could end failing tests with a `abort` so you do not have to clean up

File: crates/sui-framework/packages/bridge/sources/committee.move (hunk 1 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/committee.move:280]: probably better

**Comment by dariorussi** [comment]
> and see if your move test lib can be leveraged here

testing is fine for now, I will change things once we integrate with https://github.com/MystenLabs/sui/pull/17994

**Review by dariorussi — APPROVED** [review]
rebase should fix the snapshots/exp problem and we are good to go.
Let's rename the error I'd say

File: crates/sui-protocol-config/src/lib.rs [hunk 3]
```diff
@@ -2333,6 +2333,7 @@ impl ProtocolConfig {
                 //
                 //     // Remove a constant (ensure that it is never accessed during this version).
                 //     max_move_object_size: None,
+                50 => {}
```
Comment (dariorussi) [inline crates/sui-protocol-config/src/lib.rs:2336]: if we are indeed going to 50 please move this before the comment

**Review by dariorussi — COMMENTED** [review]
if we are in fact going to protocol version 50, please add a comment to what the version is and I'd suggest to run `./scripts/update_all_snapshots.sh` from the root of the repo (e.g. `sui`) to get all snapshots properly changed

**Pull request merged**